### ISSUE TITLE
How to configure SMS (SMS Connector) with Twilio in FreePBX?

### ISSUE THREAD

**Issue by Simoon-F** [simontelephonics/smsconnector#52]
Hello.

I installed SMS Connector and set up Twilio configuration in Provider Settings.

Then added Number in List Numbers page,
DID: 1
Users: selected 3 test numbers 1000, 1010, 1011
Provider: selected Twilio

According to the documentation, I also configured https and changed the DNS from IP access to domain name access. Do I need to copy the certificate file to the server and then set the HTTPS TLS Certificate Location and HTTPS TLS Private Key Location?

> “Allow your provider(s) webhook addresses through the FreePBX Firewall and/or your network firewall”

I have already set the port to open.

> “Set the public DNS name in the AMPWEBADDRESS setting: Advanced Settings → FreePBX Web Address”

Here I also set [voip.xxx.com](http://voip.xxx.com/).

> Have I completed the configuration here?
When I saw the description in the document: “Sending of SMS/MMS requires verification and registration performed through your provider and is outside of the scope of this module or document.”

I didn’t find where to set up Twilio’s WebHook, but I found Twilio’s SID and Token.

Then I tested it on linphone and found that a red exclamation mark would appear directly in the lower right corner of the sent text. I don’t know where else I need to configure.

![image](https://github.com/user-attachments/assets/89e8cb89-be50-4007-a04a-fdcaf81a59e6)

Do I need to configure Trunks to use SMS?

Version: FreePBX 17.0.19.1

Thanks~!

**Comment by Simoon-F** [comment]
I have configured Trunks to make outgoing calls, but I still have no idea how to send SMS messages.

**Comment by billsimon** [comment]
Please first test using the SMS widget in UCP: log in to UCP with one of the users you configured (1000, 1010, 1011) and add the SMS number to the dashboard. Once you confirm it is working in UCP then try with the SIP client. Logging including any error messages will be in /var/log/asterisk/freepbx.log

![image](https://github.com/user-attachments/assets/24e941dc-185e-4a3b-878f-c2f050edc1c6)

**Comment by Simoon-F** [comment]
@billsimon Thanks～

I encountered some new problems.

I saw in freepbx.log:

[2024-08-21 11:29:14] [freepbx.INFO]: [NOTIFICATION]-[freepbx]-[FW_TAMPERED] - You have 2 tampered files (Module "SMS Connector" signed by an invalid key.<br>Module: "SMS Connector", File: "module.sig check failed! Signed by unknown, untrusted key.") [] []

Does this mean that the https certificate file I configured needs to be stored in the server, and that I need to modify these two items in HTTPS TLS Certificate Location and HTTPS TLS Private Key Location on the Advanced Settings page?

**Comment by Simoon-F** [comment]
I seem to have solved it successfully, but it will keep prompting this. I'm not sure if this will affect my next operation, but I will continue to try.

![image](https://github.com/user-attachments/assets/510ec409-2256-46d7-81fd-ffbd762ae546)

**Comment by Simoon-F** [comment]
I can now receive and send messages in the SMS module of UCP, and I am trying to implement this function on linphone.

But I found that linphone seems to be in sip format and does not support twilio. Of course, this is just my current guess and I am still exploring.

In addition, I would like to ask, is there any software that can call and support twilio to send messages through sip tools?

Thanks～ @billsimon

**Comment by billsimon** [comment]
For SIP clients, we have successfully tested Zoiper desktop and mobile, Fanvil and Sangoma (S-series) hardphones, and Acrobits Groundwire. I played with Linphone but did not see success receiving messages to the phone. I am sure other phones may work also but we do not have a lot available to test. SIP standard phones that implement SIP MESSAGE for messaging and use only text/plain format should work fine.
